extensions/functions_string.yaml

```diff
@@ -403,3 +403,18 @@ scalar_functions:
           - value: "string"
           - value: i32
         return: "string"
+
+aggregate_functions:
+
+  -
+    name: string_agg
+    description: Concatenates a column of string values with a separator.
+    impls:
+      - args:
+          - value: "varchar<L1>"
+            name: "input"
+            description: "Column of string values."
+          - value: "varchar<L2>"
+            name: "separator"
+            description: "Separator for concatenated strings"
```

Comment:
I guess this should be marked as needing to be constant (`constant: true` IIRC), otherwise someone could pass something that varies from record to record, which makes little sense.

Reply:
Is it in the spec? I cannot find the definition for this.

Reply:
Yes,

> Constant: Whether this argument is required to be a constant for invocation. For example, in some system a regular expression pattern would only be accepted as a literal and not a column value reference.

https://substrait.io/expressions/scalar_functions/#value-argument-properties

That being said... it does appear to be missing from the schema. Huh. #305